src/scr_vm_functions.c

```diff
@@ -2995,3 +2995,78 @@ void PlayerCmd_SetSpectatedClient(scr_entref_t arg) {
 
 	gentity->client->sess.spectatorClient = Scr_GetInt(0);
 }
+
+void PlayerCmd_GetClientVersion(scr_entref_t arg) {
+	gentity_t* gentity;
+	int entityNum = 0;
+	mvabuf;
+
+	if (HIWORD(arg)) {
+		Scr_ObjectError("Not an entity");
+	} else {
+		entityNum = LOWORD(arg);
+		gentity = &g_entities[entityNum];
+
+		if (!gentity->client) {
+			Scr_ObjectError(va("Entity: %i is not a player", entityNum));
+		}
+	}
+
+    client_t *cl = &svs.clients[entityNum];
+	if (strlen(cl->xversion) < 2)
+		Scr_AddBool(0);
+	else
+		Scr_AddString(cl->xversion);
+}
+
+void GScr_getType() {
+    if( Scr_GetNumParam() != 1 ) {
+        Scr_Error( "Usage: getType( <var> )" );
+        return;
+    }
+
+    Scr_AddInt(Scr_GetType( 0 ));
```

Comment:
I'd like to see this actually return the type as a string, which would massively improve readability when using getType. Could add an alternative function (e.g. getTypeIndex or something) to get the int value of the type.
There's some more types you'll want to add in this case, I don't know all of them but these ones I can give you:
- 3 = localized string
- 9 = function